test/test.js

```diff
@@ -1,44 +1,108 @@
 'use strict';
 
 var assert = require('assert'),
-	path = require('path'),
 	gutil = require('gulp-util'),
-	testExample = require('./test-example')(),
-	content = testExample.createContents(),
-	fcount = require('./../index');
+	fcount = require('./../index'),
+    File = require('vinyl'),
+    sinon = require('sinon'),
+    _ = require('lodash'),
+    stream;
 
-describe('Should handle file contents', function (cb) {
-	var stream = fcount();
+function fakeFile(contents) {
+    return new File({
+        path: __dirname,
+        contents: new Buffer(contents)
+    });
+}
 
-	stream.write(new gutil.File({
-		path: path.join(__dirname, 'fixture.js'),
-		contents: new Buffer(content.file)
-	}));
+function processFile(contents, cb) {
+    var file = fakeFile(contents);
+    stream.write(file);
+    stream.end();
+    stream.on('finish', cb);
+}
 
-	stream.end();
+function genFunc(lineCount) {
+    var linesStr = lineCount ? _.repeat('\n', lineCount + 1) : '',
+        funcStr = '';
+    
+    switch (Math.floor(Math.random() * 2)) {
+        case 0: funcStr = 'function name() {' + linesStr + '}'; break;
+        case 1: funcStr = 'function() {' + linesStr + '}'; break;
+    }
+    return funcStr;
+}
 
-	it('Should set line25 results', function () {
-		stream.on('finish', function () {
-			assert.strictEqual(stream.result.line25, content.result.line25);
-		});
-	});
+describe('fcount', function() {        
+    beforeEach(function() {
+        stream  = fcount();
+        gutil.log = sinon.spy();
+        String.prototype.contains = function(substring) {
```

Comment:
I like the idea of doing mixed testing. I'm thinking we can primarily test the results of the counts and percentage calculations and have a describe that does a few tests on the output being logged so we have that covered as well.